src/components/CreateGameDialog.vue

```diff
@@ -1,61 +1,62 @@
 <template>
-  <v-dialog v-model="show">
-    <template #activator="{ props }">
-      <v-btn v-bind="props" color="primary" data-cy="create-game-btn" rounded elevation="8">
+  <base-dialog v-model="show" id="create-game" title="Create Game">
+    <template #activator>
+      <v-btn class="text-surface-2" color="surface-1" data-cy="create-game-btn" rounded elevation="8">
         Create Game
       </v-btn>
     </template>
-    <v-card data-cy="create-game-dialog">
-      <v-card-title>
-        <h2>Create Game</h2>
-      </v-card-title>
-      <v-card-text>
-        <div class="d-flex align-center">
-          <v-switch v-model="isRanked" label="Ranked" data-cy="create-game-ranked-switch" color="primary" />
-          <stats-scoring-dialog :show-button-text="false" />
-        </div>
-        <v-text-field
-          v-model="gameName"
-          autofocus
-          :disabled="loading"
-          label="Game Name"
-          variant="outlined"
-          data-cy="game-name-input"
-          @keydown.enter="submitNewGame"
+    <template #body>
+      <div class="d-flex align-center">
+        <v-switch
+          v-model="isRanked"
+          :label="isRanked ? 'Ranked' : 'Normal'"
+          data-cy="create-game-ranked-switch"
+          color="surface-2"
         />
-      </v-card-text>
-      <v-card-actions class="d-flex justify-end">
-        <v-btn
-          data-cy="cancel-create-game"
-          :disabled="loading"
-          variant="text"
-          color="primary"
-          @click="cancelCreateGame"
-        >
-          Cancel
-        </v-btn>
-        <v-btn
-          data-cy="submit-create-game"
-          :loading="loading"
-          color="primary"
-          variant="flat"
-          @click="submitNewGame"
-        >
-          Create Game
-        </v-btn>
-      </v-card-actions>
-    </v-card>
-  </v-dialog>
+        <stats-scoring-dialog :show-button-text="false" />
+      </div>
+      <v-text-field
+        v-model="gameName"
+        autofocus
+        :disabled="loading"
+        label="Game Name"
+        variant="outlined"
+        data-cy="game-name-input"
+        @keydown.enter="submitNewGame"
+      />
```

Comment:
This should be wrapped in an actual form, and then you won't need the `.enter` modifier. It's also more valid and accessible with the form wrapper.

Reply:
like this ? :

```vue
      <v-form>
        <v-text-field
          v-model="gameName"
          autofocus
          :disabled="loading"
          label="Game Name"
          variant="outlined"
          data-cy="game-name-input"
          @keydown="submitNewGame"
        />
      </v-form>
```

Reply:
/
```vue
      <v-form @submit.prevent="submitNewGame">
        <v-text-field
          v-model="gameName"
          autofocus
          :disabled="loading"
          label="Game Name"
          variant="outlined"
          data-cy="game-name-input"
        />
      </v-form>
``` 


```vue
        <v-btn
          type="submit"
          data-cy="submit-create-game"
          :loading="loading"
          color="surface-1"
          variant="flat"
          @click="submitNewGame"
        >
          Create Game
        </v-btn>
``` 
like this right ?

Reply:
Close, but there's a gotcha because the #actions slot separates the main content from the buttons in a way that messes up putting both the form content + the buttons inside the form.

Check the `ReauthenticateDialog` for an example of how to do this. Basically we can put an empty, hidden `<button type="submit">` inside the form just to make it submittable, and then put the actual button (as a non-submit type button) in the actions slot as usual, giving it a click handler that uses a template ref to submit the form. Then the form itself can have an @submit.prevent handler to do the actual request to create the game